src/bgw/job_stat.c

```diff
@@ -492,24 +488,15 @@ ts_bgw_job_stat_upsert_next_start(int32 bgw_job_id, TimestampTz next_start)
 	if (next_start == DT_NOBEGIN)
 		elog(ERROR, "cannot set next start to -infinity");
 
-	/* Use double-check locking */
+	Relation rel =
+		table_open(catalog_get_table_id(ts_catalog_get(), BGW_JOB_STAT), ShareRowExclusiveLock);
 	if (!bgw_job_stat_scan_job_id(bgw_job_id,
 								  bgw_job_stat_tuple_set_next_start,
 								  NULL,
 								  &next_start,
 								  RowExclusiveLock))
-	{
-		Relation rel =
-			table_open(catalog_get_table_id(ts_catalog_get(), BGW_JOB_STAT), ShareRowExclusiveLock);
-		/* Recheck while having a self-exclusive lock */
-		if (!bgw_job_stat_scan_job_id(bgw_job_id,
-									  bgw_job_stat_tuple_set_next_start,
-									  NULL,
-									  &next_start,
-									  RowExclusiveLock))
-			bgw_job_stat_insert_relation(rel, bgw_job_id, false, next_start);
-		table_close(rel, ShareRowExclusiveLock);
-	}
+		bgw_job_stat_insert_relation(rel, bgw_job_id, false, next_start);
```

Comment:
Same as above in `ts_bgw_job_stat_mark_start()`.

Reply:
Same answer as above as well.